Modules/ThirdParty/Eigen3/src/itkeigen/Eigen/Core

```diff
@@ -8,8 +8,12 @@
 // Public License v. 2.0. If a copy of the MPL was not distributed
 // with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-#ifndef EIGEN_CORE_H
-#define EIGEN_CORE_H
+#ifndef EIGEN_CORE_MODULE_H
+#define EIGEN_CORE_MODULE_H
+
+#if defined(ITK_EIGEN_MANGLE)
+#define Eigen itkeigen
+#endif
```

Comment:
```suggestion
```

My understanding is that there are no symbols in Eigen, it is a header only.

Reply:
Mangling for a header-only library means that is compatible with third-parties including their own eigen. And it won't collide with ours.

If they have in the same TU the internal itk eigen and their own eigen, both eigens might collide.
There are two things to avoid this: Change the namespace, and change the header guards.

I think both are needed, but I followed VTK and only change the namespace.

We might need a python or sed command to change the header guards.

All this mangling discussion is triggered by these 3 lines.

Reply:
Discarded